PACKAGE = "cascaded_pid_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("linear_p", double_t, 0, "p component of pid for linear vel.", 1., 0, 100)
gen.add("linear_i", double_t, 0, "i component of pid for linear vel.", 0.0, 0, 100)
gen.add("linear_d", double_t, 0, "d component of pid for linear vel.", 0.0, 0, 100)
gen.add("linear_sat", double_t, 1, "saturation of pid for linear vel.", 10.0, 0, 100)

gen.add("angular_p", double_t, 0, "p component of pid for angular vel.", 1., 0, 100)
gen.add("angular_i", double_t, 0, "i component of pid for angular vel.", 0.0, 0, 100)
gen.add("angular_d", double_t, 0, "d component of pid for angular vel.", 0.0, 0, 100)
gen.add("angular_sat", double_t, 1, "saturation of pid for angular vel.", 3.0, 0, 100)

gen.add("odom_vel_in_world", bool_t, 0, "Is odometry velocity supplied in world frame? (gazebo)", True)

exit(gen.generate(PACKAGE, "cascaded_pid_controller", "VelocityControl"))
